### PR TITLE
Argonaut=>Circe - for UI

### PR DESCRIPTION
**Pull request by mproch**
Should not change behaviour, probably some akka-http routes need some polishing...

### CODE REVIEW

File: ui/server/src/test/scala/pl/touk/nussknacker/ui/api/ProcessesExportImportResourcesSpec.scala [hunk 1]
```diff
@@ -95,18 +110,34 @@ class ProcessesExportImportResourcesSpec extends FlatSpec with ScalatestRouteTes
...
+      }
+    }
+
+  }
+
   private def assertProcessPrettyPrinted(response: String, process: CanonicalProcess): Unit = {
-    response shouldBe UiProcessMarshaller.toJson(process).spaces2
-      .replaceAll(Pattern.quote("[]"), "[ ]")
-      .replaceAll("\\{[ \n]+\\}", "{ }")
+    response.replace("\n", "").replace(" ", "") shouldBe
```
Comment (jedrz) [inline ui/server/src/test/scala/pl/touk/nussknacker/ui/api/ProcessesExportImportResourcesSpec.scala]: `response shouldBe UiProcessMarshaller.toJson(process).spaces2`? I think it's good idea to add assertions that response contains new lines.

File: engine/api/src/main/scala/pl/touk/nussknacker/engine/api/MetaData.scala [hunk 2]
```diff
@@ -24,9 +29,19 @@ object ProcessAdditionalFields {
   implicit val encoder: EncodeJson[ProcessAdditionalFields] = {
     EncodeJson.derive[ProcessAdditionalFields]
   }
+
+  //TODO: is this currently needed?
```
[the remark this answers is not in the document]
Comment (mproch) [reply engine/api/src/main/scala/pl/touk/nussknacker/engine/api/MetaData.scala]: yes, this is why I added only TODO, I think we're already after all necessary migrations in current deployments...

File: ui/server/src/test/scala/pl/touk/nussknacker/ui/api/ProcessesExportImportResourcesSpec.scala (hunk 1 above)
Comment (mproch) [reply ui/server/src/test/scala/pl/touk/nussknacker/ui/api/ProcessesExportImportResourcesSpec.scala]: true, I added sth nicer...

File: ui/server/src/test/scala/pl/touk/nussknacker/ui/api/ProcessesExportImportResourcesSpec.scala [hunk 3]
```diff
@@ -95,18 +110,34 @@ class ProcessesExportImportResourcesSpec extends FlatSpec with ScalatestRouteTes
...
+    val processToSave = ProcessTestData.sampleDisplayableProcess
+    saveProcess(processToSave) {
+      status shouldEqual StatusCodes.OK
+
+      val testSvg = "<svg viewBox=\"0 0 120 70\" xmlns=\"http://www.w3.org/2000/svg\">\n  " +
+        "<path d=\"M20,20h20m5,0h20m5,0h20\" stroke=\"#c00000\" stroke-width=\"10\"/>\n  " +
+        "<path d=\"M20,40h20m5,0h20m5,0h20M30,30v20m25,0v-20m25,0v20\" stroke=\"#008000\" stroke-width=\"6\"/>\n</svg>"
+
+      Post(s"/processesExport/pdf/${processToSave.id}/2", HttpEntity(testSvg)) ~> routeWithAllPermissions ~> check {
+        //TODO: check it's pdf?
```
[the remark this answers is not in the document]
Comment (mproch) [reply ui/server/src/test/scala/pl/touk/nussknacker/ui/api/ProcessesExportImportResourcesSpec.scala]: yes, and checking bytes (just to be sure there's no .toString or json string), I added checks

File: engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/graph/evaluatedparam.scala [hunk 4]
```diff
@@ -1,11 +1,17 @@
...
 
+import io.circe.generic.JsonCodec
 import pl.touk.nussknacker.engine.graph.expression.Expression
 
 object evaluatedparam {
 
-  case class Parameter(name: String, expression: Expression)
+  object Parameter {
+    //FIXME:
+    def tupled(tpl: (String, Expression)) = Parameter(tpl._1, tpl._2)
```
[the remark this answers is not in the document]
Comment (mproch) [reply engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/graph/evaluatedparam.scala]: yes... this is strange issue caused by @JsonCodec, I added detailed coment....

**Comment by jedrz** [comment]
We have almost finished testing this change against our project and it looks good :+1: . Approved.

**Pull request merged**